BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/BetterShardsPlugin.java

```diff
@@ -150,20 +134,23 @@ private void addPlayerToTransit(final UUID uuid, final String server){
 
 			@Override
 			public void run() {
-				transit.remove(uuid);
-				Player player = Bukkit.getPlayer(uuid);
-				if (player != null && !player.isOnline()) {
+				if (isPlayerInTransit(uuid)) {
+					info(uuid + " failed to transit, was removed by timeout");
 					PlayerFailedToTransitEvent event = new PlayerFailedToTransitEvent(uuid, server);
 					Bukkit.getPluginManager().callEvent(event);
 				}
-				else {
-					PlayerEnsuredToTransitEvent event = new PlayerEnsuredToTransitEvent(uuid, server);
-					Bukkit.getPluginManager().callEvent(event);
-				}
 			}
 			
-		}, 20);
+		}, 600); //30 seconds timeout
 	}
+	
+	public void notifySuccessfullTransfer(UUID player) {
+		if (isPlayerInTransit(player)) {
+			//TODO Readd PlayerEnsuredToTransitEvent here
+			transit.remove(player);
```

Comment:
Doesnt really matter, because the timeout will first check, whether the player is still in transit. If he isnt, it will do nothing.